src/web_interface/components/compare_routes.py

```diff
@@ -192,34 +190,24 @@ def compare_text_files(self, uid_1: str, uid_2: str, root_uid_1: str | None = No
                 error=f'Can\'t compare non-text mimetypes. ({diff_files[0].mime} vs {diff_files[1].mime})',
             )
 
-        diffstr = self._get_file_diff(*diff_files)
+        with ConnectTo(self.intercom, self._config) as intercom:
+            diff_str = intercom.get_file_diff((uid_1, uid_2))
+        if diff_str is None:
+            return render_template('compare/error.html', error='File(s) not found.')
+        diff_str = diff_str.replace('\\', '\\\\').replace('`', '\\`').replace('${', '\\$\\{')
```

Comment:
Isn't there another method to do the replacements?
Is this maintainable to replace everything ourselfes?

I had a look at the javascript which apparently I wrote some time ago and you patched a bug in it.
How does this line work? 
``` $('<textarea />').html(`{{ diffstr }}`).text(); ```
I fail to find the documentation.

Also it seems that I used backticks for enclosing the `{{ diffstr }}`. I'm unsure why. Maybe to avoid having to replace quotes in the diffstring.
I really have little experience in html/js programming but isn't there a way to do this?
Should we maybe have a hidden html element that contains the diffstring? This way we would not have to do any escaping ourselfs but jinja will do it for us.

Reply:
Actually I think I have found a robust (and maintainable) way to do this: the jinja filter "tojson" also escapes characters so passing this as `{{ diffstr | tojson }}` to the javascript code seems to work every time (at least everything that I tried that was problematic before). The `$('<textarea />')` construct was used for escaping html but it seems we also don't need that anymore.

Reply:
Good find!